developer_docs/syntactic_stage.md

```diff
@@ -0,0 +1,82 @@
+# Syntactic Stage
+
+The syntactic stage is described by the file [pyccel.parser.syntactic](../pyccel/parser/syntactic.py)
+
+The syntactic stage serves 4 main purposes:
+1.  Convert Python's [AST](https://docs.python.org/3/library/ast.html) (abstract syntax tree) representation of the python file to Pyccel's AST representation (objects of the classes in the folder [pyccel.ast](../pyccel/ast))
+2.  Raise an error for any syntax used that is not yet supported by pyccel
+3.  Convert header comments from strings to Pyccel's AST representation
```

Comment:
Can this line be used to reference the CommentLine and the CommentMultiline classes?

Reply:
I would rather do that in the header section than in the overview. But otherwise yes, it can be updated to do this. Should I point at the code given that the sphinx docs are not yet ready?

Reply:
Thank you! I think we should come up with a scheme to reference missing documentation so it can be written as soon as possible. Maybe distinguish between pointing to autogenerated doc / hand written doc. With the sphinx doc being what it is currently, i don't see how we could point to specific classes and methods. pages are generated for modules only.

Reply:
For now I am not pointing to anything that does not exist yet. Pointing to code is equally hard as you need to keep the line numbers updated.

Does the sphinx HTML not contain ids that you can link to? Maybe only once you have the settings right so everything has its own section?